Gems/Terrain/Code/Source/TerrainRenderer/TerrainClipmapManager.cpp

```diff
@@ -484,36 +481,42 @@ namespace Terrain
         }
 
         // detail clipmap data:
-        hasUpdate = false;
         for (uint32_t clipmapIndex = 0; clipmapIndex < m_detailClipmapStackSize; ++clipmapIndex)
         {
-            ClipmapBoundsRegionList updateRegionList = UpdateCenter(m_detailClipmapBounds[clipmapIndex], m_clipmapData.m_detailClipmapCenters[clipmapIndex]);
+            ClipmapBounds& clipmapBounds = m_detailClipmapBounds[clipmapIndex];
+
+            ClipmapBoundsRegionList updateRegionList = clipmapBounds.UpdateCenter(currentViewPosition);
+
+            // write updated center
+            Vector2i center = clipmapBounds.GetModCenter();
+            m_clipmapData.m_clipmapCenters[clipmapIndex][2] = aznumeric_cast<uint32_t>(center.m_x);
+            m_clipmapData.m_clipmapCenters[clipmapIndex][3] = aznumeric_cast<uint32_t>(center.m_y);
+
+            AZ::Vector2 centerWorld = clipmapBounds.GetCenterInWorldSpace();
+            m_clipmapData.m_clipmapWorldCenters[clipmapIndex][2] = centerWorld.GetX();
+            m_clipmapData.m_clipmapWorldCenters[clipmapIndex][3] = centerWorld.GetY();
 
-            for (uint32_t i = 0; i < ClipmapBounds::MaxUpdateRegions; ++i)
+            for (uint32_t i = 0; i < updateRegionList.size(); ++i)
             {
-                if (updateRegionList.size() > i)
-                {
-                    AZStd::array<uint32_t, 4> aabb = { (uint32_t)updateRegionList[i].m_localAabb.m_min.m_x,
-                                                       (uint32_t)updateRegionList[i].m_localAabb.m_min.m_y,
-                                                       (uint32_t)updateRegionList[i].m_localAabb.m_max.m_x,
-                                                       (uint32_t)updateRegionList[i].m_localAabb.m_max.m_y };
-                    m_clipmapData.m_detailClipmapBoundsRegions[clipmapIndex + ClipmapConfiguration::DetailClipmapStackSizeMax * i] = aabb;
-                }
-                else
-                {
-                    m_clipmapData.m_detailClipmapBoundsRegions[clipmapIndex + ClipmapConfiguration::DetailClipmapStackSizeMax * i] = zero;
-                }
+                AZStd::array<uint32_t, 4> aabb = { (uint32_t)updateRegionList[i].m_localAabb.m_min.m_x,
+                                                   (uint32_t)updateRegionList[i].m_localAabb.m_min.m_y,
+                                                   (uint32_t)updateRegionList[i].m_localAabb.m_max.m_x,
+                                                   (uint32_t)updateRegionList[i].m_localAabb.m_max.m_y };
```

Comment:
Same here - should use aznumeric_cast, and same question about truncation vs ceil().